assets/src/edit-story/app/media/media3p/test/useFetchMediaEffect.js

```diff
@@ -60,7 +60,7 @@ describe('useFetchMediaEffect', () => {
         useFetchMediaEffect({
           provider: 'unsplash',
           selectedProvider: 'unsplash',
-          pageToken: 'pageToken',
+          pageToken: undefined,
```

Comment:
Why was this part changed? This was testing that the current page's token is propagated through fetchMediaStart, fetchMediaSuccess etc.

Reply:
So with this current implementation, the media is only fetched if no media has been fetched (pageToken and nextPageToken are both falsey) or sth has changed from the previous render. I modified the test to ensure that the fetch would happen, and I wasn't able to figure out how to trigger the case where dependencies of a useEffect hook change. Do you know how to do that?

Reply:
I think calling renderUseFetchMediaEffect() again with a different property value for one of the dependencies will re-trigger the useEffect().

Reply:
I did, it re-triggers the useEffect but the previous props are not kept at all. I suspect because it's a completely different instance.

Reply:
Interesting, perhaps useRef is only maintained within the same act() call?

Reply:
yeah ,tried running the useFetchMediaEffect in the same act, and even inside the same renderHook, but the useRef is not maintained :(